### ISSUE TITLE
Add sudt contract/dapp as the second demo

### ISSUE THREAD

**Issue by Keith-CY** [ckb-js/kuai#250]
There's a document of **how to write an sudt script by capsule** at https://docs.nervos.org/docs/labs/sudtbycapsule/ which covers merely the contract part.

This issue aims to describe the complete workflow of writing an sudt dapp, including:
1. write an sudt script by kuai
2. build a service to mint, and manage usdt
3. deliver the document as an additional paper to the **how to write an sudt script by capsule**

**Comment by Keith-CY** [comment]
A team in the community is going to issue a meme token of sudt, which is governed by [subdid](https://www.did.id/subdid).

> Choose a specific .bit name for the meme token, such as "pepe.bit", and then distribute subdids of pepe.bit according to certain rules. The distribution logic doesn't need to be considered as the .bit community can already implement it.
> 
> Owners of "pepe.bit" addresses can synchronously issue meme tokens called "pepe".
> 
> Holders of different subdids under different rules can receive different amounts of meme tokens on different chains such as ETH/BSC/Doge chain through specified interfaces. For example, holders of 3-digit subdids can receive 100 million tokens, those with 4-digit subdids can receive 10,000 tokens, and those with 5-digit subdids can receive 100 tokens.
> 
> Users who have received meme tokens can swap them on L2 for CKB and then withdraw them to L1.

A tutorial on **how to build a dapp for sudt mint and management** is desired and should be specific for their case.

They'll start the development with this tutorial.

**Comment by Keith-CY** [comment]
> A team in the community is going to issue a meme token of sudt, which is governed by [subdid](https://www.did.id/subdid).
> 
> > Choose a specific .bit name for the meme token, such as "pepe.bit", and then distribute subdids of pepe.bit according to certain rules. The distribution logic doesn't need to be considered as the .bit community can already implement it.
> > Owners of "pepe.bit" addresses can synchronously issue meme tokens called "pepe".
> > Holders of different subdids under different rules can receive different amounts of meme tokens on different chains such as ETH/BSC/Doge chain through specified interfaces. For example, holders of 3-digit subdids can receive 100 million tokens, those with 4-digit subdids can receive 10,000 tokens, and those with 5-digit subdids can receive 100 tokens.
> > Users who have received meme tokens can swap them on L2 for CKB and then withdraw them to L1.
> 
> A tutorial on **how to build a dapp for sudt mint and management** is desired and should be specific for their case.
> 
> They'll start the development with this tutorial.

The team was unavailable last week, so there are no updates since then. Now they are ready and going to design the PRD

**Comment by Keith-CY** [comment]
No response from the community team, maybe we have to design it ourselves. The basic protocol will be added by 5/29

**Comment by Keith-CY** [comment]
### Service Part

```mermaid
flowchart TD
    minter[Minter]
    wallet[Wallet]
    tab_1[[Define new token]]
    tab_2[[Manage minted tokens]]
    tab_3[[mint]]
    tab_4[[update token info]]
    tab_5[[view distribution]]
    page_1["Fill in token info
    ------------------
    icon
    symbol
    name
    amount
    decimal
    description
    website"]
    page_2[Token List]
    page_3[Token Detail]
    page_4[Welcome page]
    page_5["Mint
    ----
    receiver
    amount
    "]
    page_6["Update token info
    -----------------
    name
    symbol
    icon
    description
    website"]
    page_7["Distribution
    ------------
    holder list
    Total minted"]

    ckb_explorer[CKB explorer]

    minter -- connect to --> wallet
    wallet -- redirect to --> page_4
    page_4 -- switch tabs --> tabs{mint / manage}
    tabs -- mint --> tab_1 
    tabs -- manage --> tab_2
    tab_1 -- show token form --> page_1 --> page_3
    page_1 -- submit token info --> ckb_explorer
    tab_2 --> page_2 --> page_3
    page_3 --> tabs_2{transfer / update token info / view distribution}
    tabs_2 -- mint --> tab_3 --> page_5
    tabs_2 -- update token info --> tab_4 --> page_6
    tabs_2 -- view distribution --> tab_5 --> page_7
```

### Client Part

```mermaid
flowchart TD
    holder[Holder]
    wallet[Wallet]
    page_1[Welcome page]
    page_2[Token List]
    page_3["Transfer
    ---------
    receiver
    amount
    "]
    page_4[CKB Explorer]


holder -- connect to --> wallet
wallet -- redirect to --> page_1
page_1 -- switch tabs --> tabs_1{Token list}
tabs_1 -- token list --> page_2
page_2 --> page_3
page_2 -- history --> page_4
```

**Comment by Keith-CY** [comment]
Halted in the community team, so I will request PMs to join and design this DApp.

**Comment by Keith-CY** [comment]
How is it going @Sven-TBD

**Comment by Sven-TBD** [comment]
> How is it going @Sven-TBD

Still working on the project background and trying the original process outlined in the tutorial. I tried the first demo(the MVP one) and am going through the interaction flow for the second demo now. We can discuss and review some user stories tomorrow to make sure the whole requirements of the second demo are confirmed.

**Comment by Keith-CY** [comment]
> > How is it going @Sven-TBD
> 
> Still working on the project background and trying the original process outlined in the tutorial. I tried the first demo(the MVP one) and am going through the interaction flow for the second demo now. We can discuss and review some user stories tomorrow to make sure the whole requirements of the second demo are confirmed.

Any update?

**Comment by Sven-TBD** [comment]
Writing the PRD from the original sketch, I can commit and push a version tonight.

**Comment by Sven-TBD** [comment]
Update the PRD branch with a draft requirement doc about this tool here
https://github.com/ckb-js/kuai/blob/52925123c49dd45de682a5291a9ae2c0637a8221/docs/PRD/user_story.md

**Comment by Keith-CY** [comment]
How is it going

**Comment by Sven-TBD** [comment]
> How is it going

Most flaws put in the last discussion have been fixed on the last commit. So in this commit, I just did some wording polishing jobs and made some modifications to the details. So I think this doc ready for tomorrow's discussion and could be finalized if there are no other requirements. And I could move on to follow other parts of kuai project. 

https://github.com/ckb-js/kuai/commit/9124d70ceac8441a84682ed79f7d0fd247616c5d

I just found I'm not able to create a new repo to put this in a new repo.
